client/client.go

```diff
@@ -511,7 +516,9 @@ func newClientWithKeyspaceName(
 		return nil, err
 	}
 	c.pdSvcDiscovery.SetKeyspaceID(c.keyspaceID)
-
+	c.serviceModeKeeper.SetKeyspaceID(c.keyspaceID)
```

Comment:
I see. We call "c.pdSvcDiscovery.SetKeyspaceID(c.keyspaceID)" after service mode already switching to API mode and tso service discovery already initialized, so here we need to set the tso_service_discovery's keyspace id too.

Reply:
ok, I also add it to comments